invokeai/backend/quantization/gguf/loaders.py

```diff
@@ -0,0 +1,68 @@
+# Largely based on https://github.com/city96/ComfyUI-GGUF
+
+from pathlib import Path
+
+import gguf
+import torch
+
+from invokeai.backend.quantization.gguf.layers import GGUFTensor
+from invokeai.backend.quantization.gguf.utils import detect_arch
+
+
+def gguf_sd_loader(
+    path: Path, handle_prefix: str = "model.diffusion_model.", data_type: torch.dtype = torch.bfloat16
+) -> dict[str, GGUFTensor]:
```

Comment:
`gguf_sd_loader()` is pretty messy (unnecessary complexity, unused vars, mixes general sd loading with model-specific key modifications, etc.). I know it's mostly copied from elsewhere, but feels like we should clean it up. I'd propose having a simple utility function for loading GGUF files, and then separate functions for key prefix handling and model-specific modifications (if we even need those for our use case).

The utility loader would be as simple as (not tested):

```python
def load_gguf_sd(path: Path) -> dict[str, GGUFTensor]:
    reader = gguf.GGUFReader(path)
    sd: dict[str, GGUFTensor] = {}
    for tensor in reader.tensors:
        torch_tensor = torch.from_numpy(tensor.data)
        shape = torch.Size(tuple(int(v) for v in reversed(tensor.shape)))
        if tensor.tensor_type in {gguf.GGMLQuantizationType.F32, gguf.GGMLQuantizationType.F16}:
            torch_tensor = torch_tensor.view(*shape)
        name = tensor.name
        sd[name] = GGUFTensor(torch_tensor, tensor_type=tensor.tensor_type, tensor_shape=shape)

    return sd
```